pkg/bgpv1/manager/reconcile.go

```diff
@@ -381,6 +385,210 @@ func (r *ExportPodCIDRReconciler) Reconcile(ctx context.Context, p ReconcilePara
 	return nil
 }
 
+type PodIPPoolReconcilerOut struct {
+	cell.Out
+
+	Reconciler ConfigReconciler `group:"bgp-config-reconciler"`
+}
+
+type PodIPPoolReconciler struct {
+	poolStore DiffStore[*v2alpha1api.CiliumPodIPPool]
+}
+
+func NewPodIPPoolReconciler(poolStore DiffStore[*v2alpha1api.CiliumPodIPPool]) PodIPPoolReconcilerOut {
+	if poolStore == nil {
+		return PodIPPoolReconcilerOut{}
+	}
+
+	return PodIPPoolReconcilerOut{
+		Reconciler: &PodIPPoolReconciler{
+			poolStore: poolStore,
+		},
+	}
+}
+
+func (r *PodIPPoolReconciler) Priority() int {
+	return 50
+}
+
+func (r *PodIPPoolReconciler) Reconcile(ctx context.Context, p ReconcileParams) error {
+	lp := r.populateLocalPools(p.CiliumNode)
+
+	if err := r.fullReconciliation(ctx, p.CurrentServer, p.DesiredConfig, lp); err != nil {
+		return fmt.Errorf("full reconciliation failed: %w", err)
+	}
+
+	return nil
+}
+
+// populateLocalPools returns a map of allocated multi-pool IPAM CIDRs of the local CiliumNode,
+// keyed by the pool name.
+func (r *PodIPPoolReconciler) populateLocalPools(localNode *v2api.CiliumNode) map[string][]netip.Prefix {
+	if localNode == nil {
+		return nil
+	}
+
+	lp := make(map[string][]netip.Prefix)
+	for _, pool := range localNode.Spec.IPAM.Pools.Allocated {
+		var prefixes []netip.Prefix
+		for _, cidr := range pool.CIDRs {
+			if p, err := cidr.ToPrefix(); err == nil && (p.Addr().Is4() || p.Addr().Is6()) {
```

Comment:
Out of curiosity: Why is the `(p.Addr().Is4() || p.Addr().Is6())` check needed? Is there case where an IP is neither?

Reply:
Thanks for catching this. Yes, these checks are unneeded and removed in commit b18c075.